zcash_client_backend/src/data_api.rs

```diff
@@ -290,22 +286,46 @@ impl AccountBalance {
     }
 
     /// Returns the total value of unspent transparent transaction outputs belonging to the wallet.
+    #[deprecated(
+        note = "this function is deprecated. Please use AccountBalance::unshielded_balance instead."
+    )]
     pub fn unshielded(&self) -> NonNegativeAmount {
-        self.unshielded
+        self.unshielded_balance.total()
+    }
+
+    /// Returns the [`Balance`] of unshielded funds in the account.
+    pub fn unshielded_balance(&self) -> &Balance {
+        &self.unshielded_balance
+    }
+
+    /// Provides a `mutable reference to the [`Balance`] of transparent funds in the account
+    /// to the specified callback, checking invariants after the callback's action has been
+    /// evaluated.
+    pub fn with_unshielded_balance_mut<A, E: From<BalanceError>>(
+        &mut self,
+        f: impl FnOnce(&mut Balance) -> Result<A, E>,
+    ) -> Result<A, E> {
+        let result = f(&mut self.unshielded_balance)?;
+        self.check_total()?;
+        Ok(result)
     }
 
     /// Adds the specified value to the unshielded total, checking for overflow of
     /// the total account balance.
+    #[deprecated(
+        note = "this function is deprecated. Please use the `Balance::add_spendable_value` on the unshielded field instead instead."
+    )]
     pub fn add_unshielded_value(&mut self, value: NonNegativeAmount) -> Result<(), BalanceError> {
```

Comment:
It is unclear whether we should retain this method at all. It may be better to make a breaking change and delete it. The problem is that we cannot know what the intent was, because previously only the total was tracked. The value to be added might be spendable, pending change, pending non-change, or some combination.

Option 1: assume that the value to be added is pending non-change, and document the assumption.
```suggestion

    /// Adds the specified value to the pending non-change unshielded balance,
    /// checking for overflow of the total account balance.
    ///
    /// Note: Previously only the *total* unshielded balance was tracked. In this
    /// implementation, we assume that the intent was to add the value to the
    /// unshielded balance as pending non-change. This assumption may or may not
    /// be correct.
    #[deprecated(
        note = "this function is deprecated. Please use `[AccountBalance::with_unshielded_balance_mut]` with a closure that calls the appropriate `add_*_value` method(s) of [`Balance`] on its argument."
    )]
    pub fn add_unshielded_value(&mut self, value: NonNegativeAmount) -> Result<(), BalanceError> {
```

Option 2 is to delete this method, which requires a breaking semver version update:
```suggestion
```

Reply:
I'm good either way. I decided to maintain functionality (proved by passing the existing tests) because it's what a non maintainer contributor should do. But I can proceed with the option you folks consider more appropriate.

Reply:
I would prefer to delete it. @str4d or @nuttycom, what do you think?

Reply:
I opted to remove it. :)